Add new Dynamic Time calculation

Abban opened the pull request:
This adds a new time display item to be used in the banner content

Ticket: https://phabricator.wikimedia.org/T351623

To acceptance test this you'll need to add a ref in a banner content component that can be updated periodically:

```js
const {
	currentDayName,
	currentDate,
	currentTime
}: DynamicContent = inject<DynamicContent>( 'dynamicCampaignText' );

const time = ref<string>( currentTime() );

setInterval( () => {
	time.value = currentTime();
}, 1000 );
```

Abban committed:
Add new Dynamic Time calculation

This adds a new time display item to be used in the banner content

Ticket: https://phabricator.wikimedia.org/T351623

gbirke commented on src/utils/DynamicContent/generators/CurrentTime.ts:
```diff
@@ -0,0 +1,23 @@
+	}
+
+	public getText( date: Date ): string {
+		return this._translator.translate( 'date-month-time-' + ( date.getMonth() + 1 ), {
```
The name of the class is a bit off from the implementation, which returns current date and time. I'm wondering if this isn't too much duplication and we'd be better (in terms of flexibility and DRY-ness) doing `{{currentDate}}, {{currenTime}}` in the banners instead of `{{ currentTime }}`. Although that would lead to an edge case at midnight if we don't update both date and time.

Abban replied:
I was thinking about that name already, lets do `currentDateAndTime`?

gbirke commented on src/utils/DynamicContent/DynamicContent.ts:
```diff
@@ -3,6 +3,7 @@ import { DynamicProgressBarContent } from '@src/utils/DynamicContent/DynamicProg
 export interface DynamicContent {
 	currentDayName: string;
 	currentDate: string;
+	currentTime: () => string;
```
Is there a reason why this needs to be method instead of a getter?

Abban replied:
When it's a getter js seems to cache the value, so it wasn't updating.

gbirke commented on src/utils/DynamicContent/DynamicCampaignText.ts:
```diff
@@ -32,6 +34,7 @@ export default class DynamicCampaignText implements DynamicContent {
 		this._campaignParameters = campaignParameters;
 		this._impressionCount = impressionCount;
 		this._cache = new Map<string, string>();
+		this.currentTime = this.currentTime.bind( this );
```
Why is this needed?

Abban replied:
Because we're using a class, and the method uses `this` internally, so we need to bind the method to the current scope when it's constructed. Otherwise we'll get 'this is not defined' errors.

gbirke commented on src/utils/DynamicContent/DynamicContent.ts:
```diff
@@ -3,6 +3,7 @@ import { DynamicProgressBarContent } from '@src/utils/DynamicContent/DynamicProg
 export interface DynamicContent {
 	currentDayName: string;
 	currentDate: string;
+	currentDateAndTime: () => string;
```
It would be cool if you could document the difference between properties and getter methods in the interface docbloc. Also maybe call this `getCurrentDateAndTime()`

Abban committed:
Add docblock explaining dynamic content methods

Also renames the currentDateAndTime method to getCurrentDateAndTime

Ticket: https://phabricator.wikimedia.org/T351623
